src/api/posts.ts

```diff
@@ -0,0 +1,47 @@
+import { Comment, Post, NewComment } from '../react-app-env';
+
+export const BASE_URL = 'https://mate.academy/students-api';
+
+export const getUserPosts = (userId: string) => {
+  if (userId === '0') {
+    return fetch(`${BASE_URL}/posts/`)
+      .then(response => response.json());
+  }
+
+  return fetch(`${BASE_URL}/posts/?userId=${userId}`)
+    .then(response => response.json());
+};
```

Comment:
Will be better to create separate function for getting all posts and call it here